contracts/strategy/BnbxYieldConverterStrategy.sol

```diff
@@ -0,0 +1,245 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts-upgradeable/utils/AddressUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/IERC20Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/utils/SafeERC20Upgradeable.sol";
+
+import "../masterVault/interfaces/IMasterVault.sol";
+import "../bnbx/interfaces/IStakeManager.sol";
+import "./BaseStrategy.sol";
+
+contract BnbxYieldConverterStrategy is BaseStrategy {
+
+    using SafeERC20Upgradeable for IERC20Upgradeable;
+
+    IERC20Upgradeable private _bnbxToken;
+    IStakeManager private _stakeManager;
+    IMasterVault private _vault;
+
+    struct UserWithdrawRequest {
+        address recipient;
+        uint256 amount;
+        uint256 bnbxAmount;
+        uint256 triggerTime;
+    }
+    mapping(uint256 => UserWithdrawRequest) private _withdrawRequests;
+    uint256 private _firstWithdrawIdx;
+    uint256 private _nextWithdrawIdx;
+
+    uint256 private _bnbxHoldingBalance; // amount of bnbx held by this strategy
+    uint256 private _bnbDepositBalance; // amount of bnb deposited by this strategy
+    uint256 private _bnbxToUnstake; // amount of bnbx to withdraw from stader in next batchWithdraw
+    uint256 private _bnbToDistribute; // amount of bnb to distribute to users who unstaked
+
+    event StakeManagerChanged(address stakeManager);
+
+    /// @dev initialize function - Constructor for Upgradable contract, can be only called once during deployment
+    /// @param destination Address of the stakeManager contract
+    /// @param feeRecipient Address of the fee recipient, which receives yield
+    /// @param bnbxToken Address of BNBx token
+    /// @param masterVault Address of the masterVault contract
+    /// @param stakeManager Address of stakeManager contract
+    function initialize(
+        address destination,
+        address feeRecipient,
+        address bnbxToken,
+        address masterVault,
+        address stakeManager
+    ) public initializer {
+        __BaseStrategy_init(destination, feeRecipient);
+
+        _bnbxToken = IERC20Upgradeable(bnbxToken);
+        _stakeManager = IStakeManager(stakeManager);
+        _vault = IMasterVault(masterVault);
+
+        _bnbxToken.approve(stakeManager, type(uint256).max);
+    }
+
+    /**
+     * Modifiers
+     */
+    modifier onlyVault() {
+        require(msg.sender == address(_vault), "!vault");
+        _;
+    }
+
+    /// @dev deposits the given amount of BNB into Stader stakeManager through bnbxRouter
+    function deposit() external payable onlyVault returns (uint256) {
+        uint256 amount = msg.value;
+        require(amount <= address(this).balance, "insufficient balance");
+
+        return _deposit(amount);
+    }
+
+    /// @dev deposits all the available BNB into Stader stakeManager through bnbxRouter
```

Comment:
...available BNB on router into stader ...

Reply:
missed changing comments
should be
`/// @dev deposits all the available BNB in this strategy contract to Stader stakeManager`

Will check with Helio devs as well for its use case.
as it will deposit back the users funds. (unstaked funds)
Accordingly will have to update `_bnbToDistribute `

`_bnbToDistribute  = _bnbToDistribute - address(this).balance + msg.value `

Reply:
done